### PR TITLE
fix(tokens): show sub-cent token prices instead of $0.00

### LINKED ISSUE #5088: [ADD] Show token prices lower then $0.00 (example: LUNC)
# [Add] [ADD] Show token prices lower then $0.00 (example: LUNC)

## Problem
The user wants to see token prices lower than $0.00, for example, tokens like $LUNC.

## Expected Behavior
N/A

## Solution
TBD

### Must Do
- [ ] Implement the fix/feature described above

### Must NOT Do
- Don't change unrelated code
- Don't refactor surrounding logic

## Acceptance Criteria
- [ ] `xcodebuild -scheme Vultisig build` succeeds
- [ ] Behavior matches expected behavior above

## Context
- Reported by **Jatinkkalra** (@jatinkkalra on Discord) on Discord

cc @realpaaao

### PR DESCRIPTION
**Pull request by aminsato**
## Summary

Per-token prices below one currency sub-unit (e.g. **LUNC ~$0.00006**, **USTC ~$0.0054**) were displayed as `$0.00`, hiding the real price from users. Closes #5088.

### Root cause
The unit price was rounded away **before it ever reached the UI**:
1. In `BalanceRepository`, the price `FiatValue` was built with `price.scaledFor(currency)` — `setScale(2, RoundingMode.DOWN)` for USD — so `$0.000059` became `0.00` at the data source.
2. The display mapper then formatted to the currency's standard 2 fraction digits anyway.

### Fix
- **`BalanceRepository`** — stop pre-scaling the **unit price** at all 5 construction sites; it now carries full precision. The **holdings** value (`fiatValue`) keeps `scaledFor` — a tiny holding correctly still shows `$0.00`.
- **`FiatValueToStringMapper`** — new opt-in `asPrice` mode that extends precision just far enough to reveal a sub-cent price's leading significant figures (2 sig figs, capped at 8 fraction digits).
- Apply `asPrice = true` at the per-token price display sites (`ChainTokensViewModel`, `TokenDetailViewModel`, `CustomTokenViewModel`).

Prices ≥ one sub-unit are unaffected; balances are unaffected.

## Before / After

| Before | After |
|--------|-------|
| before | after |

LUNC `$0.00` → `$0.000059`, USTC `$0.00` → `$0.0054`. Tokens with no/zero price (UUSDC, KRT) correctly remain `$0.00`.

## Test plan
- [x] `:data:compileDebugKotlin` + `:app:compileDebugKotlin`
- [x] New `FiatValueToStringMapperImplTest` cases: LUNC-style sub-cent reveal, 2-sig-fig, 8-digit cap, fallback ≥ 1¢, zero, `asFee` precedence
- [x] `AccountsRepositoryImplTest`, `ChainTokensViewModelTest` green
- [x] ktfmt clean
- [x] Device-verified on Terra Classic vault (see Before/After)

## Summary by CodeRabbit

* **New Features**
  * Improved token price formatting across token lists, token details, and custom token search, showing meaningful fractional pricing for very small values.

* **Bug Fixes**
  * Fixed incorrect rendering of sub-cent token prices by preserving full-precision values until display formatting is applied.
  * Updated formatting logic to handle price vs. fee modes correctly, including correct precision and rounding behavior.

* **Tests**
  * Expanded mapper test coverage for `asPrice` formatting (including edge cases and precedence with fee formatting).

### CODE REVIEW

File: app/src/test/java/com/vultisig/wallet/ui/models/mappers/FiatValueToStringMapperImplTest.kt [hunk 1]
```diff
@@ -92,4 +92,40 @@ internal class FiatValueToStringMapperImplTest {
...
+        // Two significant figures would need 9 digits; capped to 8 → rounds to $0.00000012.
+        mapper(FiatValue(BigDecimal("0.000000123"), "USD"), asPrice = true) shouldBe "$0.00000012"
+    }
+
+    @Test
+    fun `asPrice falls back to standard formatting at or above one cent`() = runTest {
+        val value = FiatValue(BigDecimal("1.50"), "USD")
+        mapper(value, asPrice = true) shouldBe mapper(value)
+        mapper(value, asPrice = true) shouldBe "$1.50"
+    }
```
[the remark this answers is not in the document]
Comment (aminsato) [reply app/src/test/java/com/vultisig/wallet/ui/models/mappers/FiatValueToStringMapperImplTest.kt:126]: Good catch — added `asPrice falls back to standard formatting at exactly one cent` (0.01 → $0.01) in 376875079, keeping the above-threshold 1.50 case alongside it.

**Review by rkokhatskyi — APPROVED** [review]
LGTM
